Content.Server/StationEvents/Events/NinjaSpawnRule.cs

```diff
@@ -0,0 +1,52 @@
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Ninja.Systems;
+using Content.Server.Station.Components;
+using Content.Server.StationEvents.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Random;
+using System.Linq;
+
+namespace Content.Server.StationEvents.Events;
+
+/// <summary>
+/// Event for spawning a Space Ninja mid-game.
+/// </summary>
+public sealed class NinjaSpawnRule : StationEventSystem<NinjaSpawnRuleComponent>
+{
+    [Dependency] private readonly SpaceNinjaSystem _ninja = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    protected override void Started(EntityUid uid, NinjaSpawnRuleComponent comp, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, comp, gameRule, args);
+
+        if (!TryGetRandomStation(out var station))
+            return;
+
+        var stationData = Comp<StationDataComponent>(station.Value);
+
+        // find a station grid
+        var gridUid = StationSystem.GetLargestGrid(stationData);
+        if (gridUid == null || !TryComp<MapGridComponent>(gridUid, out var grid))
+        {
+            Sawmill.Warning("Chosen station has no grids, cannot spawn space ninja!");
+            return;
+        }
```

Comment:
should prob use the GetStations() method?

Reply:
TryFindRandomTile has similar code but fails silently if it chooses an invalid one